### PR TITLE
My work about the choose token life and desabeling users 

### PR DESCRIPTION
**Pull request by axl-pis**

### CODE REVIEW

File: server-nodejs/extensions/jwt.js [hunk 1]
```diff
@@ -28,6 +28,17 @@ module.exports = function (app) {
         } else {
            obj = {'email' : req.body.username};
         }
+        let jwtRegex = RegExp('^[0-9]*[a-z]\$');
+        let jwtUnlimited = false;
+        let jwtExp = { expiresIn: conf.exp*60 };
+        if ("undefined" != req.body.jwtTime) {
```
Comment (axl-pis) [inline server-nodejs/extensions/jwt.js]: The undefined test avoid server error when it read "req.body.jwtTime"

File: server-nodejs/extensions/jwt.js [hunk 2]
```diff
@@ -28,6 +28,17 @@ module.exports = function (app) {
...
+            if ("true" === req.body.jwtTime) {
+                jwtUnlimited = true;
```
Comment (axl-pis) [inline server-nodejs/extensions/jwt.js]: To have a token with unlimited lifespan, the user must inform "true" in damas.signIn(). That's why, there is a specifically for that.

File: server-nodejs/extensions/jwt.js [hunk 3]
```diff
@@ -28,6 +28,17 @@ module.exports = function (app) {
...
+            } else if (jwtRegex.test(req.body.jwtTime) && !nameRegex.test(req.body.jwtTime)) {
+                jwtUnlimited = req.body.jwtTime;
+                jwtExp = { expiresIn: req.body.jwtTime };
+            }
+        }
```
Comment (axl-pis) [inline server-nodejs/extensions/jwt.js]: To set token lifespan, the user must inform a lifetime like "1d" or "1h". That's why, the test with regular expression stop all other string.

File: server-nodejs/extensions/jwt.js [hunk 4]
```diff
@@ -90,11 +106,21 @@ module.exports = function (app) {
...
+           if (err || null === user[0]) {
+                req.user = {};
+                return next();
+            }
+            if (user["disable"]) {
+                req.user = {};
+                return next();
+            }
+            let ignoreExp;
+            if ("true" != user[0]['unlimitJwt']){
```
Comment (remyla) [inline server-nodejs/extensions/jwt.js]: string or boolean?

**Pull request closed without merging**